correlation_context/HTTP_HEADER_FORMAT.md

```diff
@@ -10,24 +10,64 @@ A correlation context header is used to pass the name-value context properties f
 
 `Correlation-Context`
 
-Multiple correlation context headers are allowed. Values can be combined in a single header according to the [RFC 7230](https://tools.ietf.org/html/rfc7230#page-24).
-
 ## Header value
 
-`name1=value1[;properties1],name2=value2[;properties2]`
+The header value contains a list of key/value pairs. The first pair MUST contain the version of the header
+as `v=<version>`.
+
+The header MAY be sent or received as multiple header fields. Multiple header fields MUST be handled as specified by <a data-cite='!RFC7230#field.order'>RFC7230 Section 3.2.2 Field Order</a>. The header SHOULD be sent as a single field when possible, but MAY be split into multiple header fields. When sending the header as multiple header fields, it MUST be split according to <a data-cite='!RFC7230#field.order'>RFC7230</a>. When receiving multiple header fields, they MUST be combined into a single header according to <a data-cite='!RFC7230#field.order'>RFC7230</a>.
+
+When multiple headers are combined as described before, the resulting combined header will contain
+a version identifier for each individual header that was concatenated. This is intentional and parsers MUST NOT assume that any version field applies to the entire header value. After the first version identifier, another version identifier might occur at any time. Subsequent key/value pairs MUST be compliant with the denoted version until another version identifier is encountered or the end of the header value is reached.
+
+
+This section uses the Augmented Backus-Naur Form (ABNF) notation of [[!RFC5234]], including the DIGIT rule in <a data-cite='!RFC5234#appendix-B.1'>appendix B.1 for RFC5234</a>. It also includes the `OWS` rule from <a data-cite='!RFC7230#whitespace'>RFC7230 section 3.2.3</a>.
+
+The `DIGIT` rule defines numbers `0`-`9`.
+
+The `OWS` rule defines an optional whitespace character. To improve readability, it is used where zero or more whitespace characters might appear.
+
+The caller SHOULD generate the optional whitespace as a single space; otherwise, a caller SHOULD NOT generate optional whitespace. See details in the <a data-cite='!RFC7230#whitespace'>corresponding RFC</a>.
```

Comment:
"otherwise" compared to what? Why not say caller always SHOULD NOT generate whitespace? Consume liberally, produce conservatively.